app/main.py

```diff
@@ -1,34 +1,55 @@
-class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+class Ship:
+    def __init__(self, start: tuple[int], end: tuple[int]) -> None:
+        self._count_alive_decks = (end[0] - start[0]) + (end[1] - start[1]) + 1
+        self._fired_decks = set()
 
+    def fire(self, row: int, column: int) -> str:
+        cell = (row, column)
 
-class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+        if cell in self._fired_decks:
+            return "Miss!"
+
+        self._fired_decks.add(cell)
+        self._count_alive_decks -= 1
+        return {0: "Sunk!"}.get(self._count_alive_decks, "Hit!")
```

Comment:
![image](https://user-images.githubusercontent.com/108597725/195188225-dba10e1e-1055-4f5b-8396-b6a31cd9af46.png)

An interesting approach.
I can see no advantages over ternary operator in this case, though.